components/components-salesforce/components-salesforce-definition/src/main/java/org/talend/components/salesforce/tsalesforcebulkexec/TSalesforceBulkExecProperties.java

```
@@ -43,6 +45,8 @@
 
     public Property<Boolean> outputUpsertKey = newBoolean("outputUpsertKey");
 
+    public Property<ContentType> contentType = new EnumProperty(ContentType.class, "contentType");
```

did you try to hide JSON type for bulkv2 ? not sure we can do it on the UI side.
or we can throw a runtime exception for the customer.

---

bulkv2 does not support JSON format. We can only use it in bulk v1.

---

Thanks, @jzhao-talend . I have modified the code to hide ContentType for bulkV2.